panel/panel.py

```diff
@@ -207,107 +185,115 @@ def fill_panel(window, update=False, **panel_filter):
     else:
         settings.set("sublime_linter_panel_filter", panel_filter)
 
-    to_render = RenderLines()
-    for vid, view_dict in errors.items():
-        if util.is_none_or_zero(view_dict["we_count_view"]):
-            continue
-
-        view_lines = get_view_lines(view_dict, panel_filter)
-        if view_lines:
+    to_render = []
+    for bid, buf_errors in errors.items():
+        buf_lines = get_buf_lines(buf_errors, panel_filter)
```

Comment:
I think you can also filter in get_window_raw_errors in one go, unless I'm mistaken. The point is to get the right data set as soon as possible.  Then the following `if` branch would also be always true, thus unnecessary.